db/migrations/20200203102132_create_bid_event_table.sql

```diff
@@ -0,0 +1,239 @@
+-- +goose Up
+CREATE TABLE maker.bid_event
+(
+    bid_id           NUMERIC     NOT NULL,
+    contract_address TEXT        NOT NULL,
+    act              api.bid_act NOT NULL,
+    lot              NUMERIC DEFAULT NULL,
+    bid_amount       NUMERIC DEFAULT NULL,
+    ilk_identifier   TEXT    DEFAULT NULL,
+    urn_identifier   TEXT    DEFAULT NULL,
```

Comment:
Yeah I def agree that it's a bummer to negate the filtering flexibility we'd get by exposing the table directly. Which is why I think it would be ideal if we could just expose it, and somehow inform the end-user about the data's weirdness.

Although, on some level I think it's kind of self-explanatory, since nobody would really include the `ilk_identifier` or `urn_identifier` fields in a graphql query where they only wanted flop or flap events. And if they did accidentally include those columns in the query, I don't think it would be that confusing when they came back null, assuming a base level of knowledge about the different auction types. 🤷‍♂

Reply:
I think we're planning to expose all three schemas in the API, if that's relevant? But agreed that having `null` for `ilk_identifier` and `urn_identifier` doesn't seem that confusing if you're querying against a contract address for flap/flop 🤷‍♂

Reply:
One thing that's got me thinking, I wonder if we'll need to communicate anything about the need to use checksum/non-checksum addresses when querying by urn identifier 🤔 

The column is case insensitive iirc, but idk if that means you'll get results if you pass in a string with different casing than whatever's in that row?

Reply:
Good to know we'll be exposing all 3 schemas! In that case I guess we'd have to explicitly omit the query if we want to hide it from users, although I'm inclined to just leave it as is.

And that's a really good point about case-sensitivity, it hadn't really occurred to me. Our `GetOrCreateAddress` function always inserts checksummed addresses, so we should probably make sure people are aware of that. Maybe it would also be worth adding case-insensitivity to the `urn_bid_events` query; that shouldn't take much effort.

More broadly, maybe we should consider making our trigger-tables conform to a simpler standard (e.g. all lower-case addresses), to make it easier on the user. In fact, I wonder if postgraphile has a "case-insensitive" plugin?

Reply:
Yeah, I think conforming to a single standard would be a definite win. I think I'd probably lean toward checksum version rather than all lower case because you get the former when you copy an address on Etherscan, but this may also be a good place to pose the question to stakeholders about which they prefer if we generally need to choose one